src/main/java/frc/robot/Autonomous.java

```diff
@@ -0,0 +1,168 @@
+package frc.robot;
+
+import com.revrobotics.RelativeEncoder;
+// import com.revrobotics.SparkMaxRelativeEncoder;
+
+// SUBSYSTEM IMPORTS
+import frc.robot.subsystems.BallPath.BallPath.BallAction;
+import frc.robot.subsystems.BallPath.Shooter.Shooter.ShotPosition;
+import frc.robot.subsystems.BallPath.BallPathImpl;
+import frc.robot.subsystems.Drivetrain.DriveImpl;
+
+import com.revrobotics.SparkMaxPIDController;
+// import com.revrobotics.CANSparkMax.ControlType;
+
+
+import edu.wpi.first.math.controller.PIDController;
+
+// import java.util.concurrent.TimeUnit;
+// import ca.team3161.lib.robot.TitanBot;
+
+public class Autonomous {
+
+    private DriveImpl drivetrain;
+    // private Shooter shooter;
+    // private Intake intake;
+    // private Elevator elevator;
+    private BallPathImpl ballPath;
+    private double wheelCircumference = Math.pow((Math.PI*2), 2);
+    private double distance;
+    private double setPoint;
+    private double max = 1000;
+    private double min = 0;
+
+    private double kP = 0.00001;
+    private double kI = 0;
+    private double kD = 0;
+
+    private SparkMaxPIDController leftPIDController;
+    private SparkMaxPIDController rightPIDController;
+
+    private RelativeEncoder leftEncoder;
+    private RelativeEncoder rightEncoder;
+
+    // private double maxSpeed = 1;
+    private double zRotation = 0;
+    private double targetDistance;
+
+    private PIDController positionPIDController;
+
+
+
+
+
+    public Autonomous(DriveImpl drivetrain, BallPathImpl ballPath){
```

Comment:
This should only need a `Drive` and ` BallPath` - the interfaces, not the classes. It should only interact with them by calling the interface methods and not by directly accessing any of their internals - that's the point of using interfaces, so that there is clean separation of concerns.